api/src/nv_ingest_api/util/dataloader/dataloader.py

```diff
@@ -0,0 +1,177 @@
+
+logger = logging.getLogger(__name__)
+
+if importlib.util.find_spec("ffmpeg") is None:
+    logger.error(
+        "Unable to load the Dataloader,ffmpeg was not installed, "
+        "please install it using `pip install ffmpeg-python` and `apt-get install ffmpeg`"
+    )
+else:
+    import ffmpeg
+
+
+class LoaderInterface(ABC):
+
+    @abstractmethod
+    def split(self, input_path: str, output_dir: str, split_interval: int = 0):
+        pass
+
+
+def _probe(filename="pipe:", cmd="ffprobe", format=None, file_handle=None, timeout=None, **kwargs):
+    args = [cmd, "-show_format", "-show_streams", "-of", "json"]
+    args += ffmpeg._utils.convert_kwargs_to_cmd_line_args(kwargs)
+    args += ["pipe:"]
+
+    p = subprocess.Popen(args, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    communicate_kwargs = {}
+    if timeout is not None:
+        communicate_kwargs["timeout"] = timeout
+    if file_handle:
+        communicate_kwargs["input"] = file_handle
+    out, err = p.communicate(**communicate_kwargs)
+    if p.returncode != 0:
+        raise ffmpeg._run.Error("ffprobe", out, err)
+    return json.loads(out.decode("utf-8"))
+
+
+class MediaInterface(LoaderInterface):
+
+    def split(self, input_path: str, output_dir: str, split_interval: int = 0):
+        """
+        Split a video into smaller chunks of `split_interval` size.
+        input_path: str, path to the video file
+        output_dir: str, path to the output directory
+        split_interval: int, size of the chunk in MB
+        """
+        import ffmpeg
+        from upath import UPath as Path
+
+        path_file = Path(input_path)
+        file_name = path_file.stem
+        suffix = path_file.suffix
+        output_dir = Path(output_dir)
+        output_dir.mkdir(parents=True, exist_ok=True)
+        output_pattern = output_dir / f"{file_name}_chunk_%04d{suffix}"
+        num_splits = 0
+        try:
+            with fsspec.open(input_path, "rb", encoding="utf-8") as f_in:
+                probe = _probe("pipe:", format=suffix, file_handle=f_in.read())
+                bitrate = float(probe["format"]["bit_rate"])
+                file_size = path_file.stat().st_size
+                duration = (file_size * 8) / bitrate
+                num_splits = math.ceil(file_size / (split_interval * 1e6))
+                segment_time = math.ceil(duration / num_splits)
+                f_in.seek(0)
+                (
+                    ffmpeg.input(
+                        "pipe:",
+                    )
+                    .output(str(output_pattern), f="segment", segment_time=segment_time, c="copy", map="0")
+                    .run(capture_stdout=True, capture_stderr=True, input=f_in.read())
+                )
+        except ffmpeg.Error as e:
+            logging.error(f"FFmpeg error for file {input_path}: {e.stderr.decode()}")
+        files = [str(output_dir / f"{file_name}_chunk_{i:04d}{suffix}") for i in range(int(num_splits))]
```

Comment:
Need some way to default, maybe optionally, clean up any files created by this process when the DataLoader goes out of scope, or we encounter some kind of unexpected processing error.

Reply:
If we do this, we would have to redo the work of splitting the files. I will add a switch that allows for clean up, but can be turned off. I can see a user having a ton of files and going through the process of breaking them up, running an iteration of the pipeline and then maybe wanting to tweak something in the pipeline but run with the same files. It would be cool if they could just use the artifacts already created and point to that directory for ingestor files. Thoughts?